lib/Runtime/PlatformAgnostic/Platform/Common/UnicodeText.ICU.cpp

```diff
@@ -266,9 +266,9 @@ namespace PlatformAgnostic
             return u_hasBinaryProperty(ch, UCHAR_ID_CONTINUE);
         }
 
-        int LogicalStringCompare(const char16* string1, const char16* string2)
+        int LogicalStringCompare(const char16* string1, int str1size, const char16* string2, int str2size)
         {
-            return PlatformAgnostic::UnicodeText::Internal::LogicalStringCompareImpl(string1, string2);
```

Comment:
We may want to have this definition be `#ifndef _WIN32` like GetLegacyCharacterClassificationType above, since even when we are using ICU on Windows, this function is trying to emulate CompareStringEx. This isn't an issue you introduced but it would be nice to fix it while this code is being touched. If you want to fix this, you will have to move the LogicalStringCompare definition in Platform\Windows\UnicodeText.cpp as well so it gets compiled in on Windows.